syntax = "proto3";
package plugin;





/*
ListPluginAgentsV1请求
*/
message ListPluginAgentsV1Request {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页数据量
    */
    int32 pageSize = 2 ;
    /*
    ID
    */
    string id = 3 ;
}

/*
ListPluginAgentsV1返回
*/
message ListPluginAgentsV1Response {
message List {
    /*
    Agent的ID
    */
    string agentId = 1 ;
    /*
    Agent的IP
    */
    string ip = 2 ;
    /*
    Agent的状态
    */
    string status = 3 ;
    /*
    插件的部署路径
    */
    repeated string deployPath = 4 ;
    /*
    部署的插件版本名称
    */
    string versionName = 5 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListPluginAgentsV1Api返回
*/
message ListPluginAgentsV1ResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    plugin.ListPluginAgentsV1Response data = 4 ;
}

